### ISSUE TITLE
disable rgw service failed when purge cluster by ceph-ansible

### ISSUE THREAD

**Issue by xiaohh2016** [ceph/ceph-ansible#5714]
**Bug Report**

What happened:
When I use the infrastructure-playbooks/purge-container-cluster.yml playbook to purge a Ceph cluster, it report an error when
disable the rgw service. The error info was list below：
```
fatal: [ceph01]: FAILED! => 
  msg: |-
    The task includes an option with an undefined variable. The error was: 'ansible.utils.unsafe_proxy.AnsibleUnsafeText object' has no attribute 'instance_name'
  
    The error appears to be in '/root/ceph-ansible/infrastructure-playbooks/purge-container-cluster.yml': line 211, column 7, but may
    be elsewhere in the file depending on the exact syntax problem.
  
    The offending line appears to be:
  
  
        - name: disable ceph rgw service
          ^ here
```
The task info was listed below:
```
    - import_role:
        name: ceph-facts
        tasks_from: set_radosgw_address

    - debug:
        msg: "{{ item }}"
      with_items: "{{ rgw_instances }}"

    - name: disable ceph rgw service
      service:
        name: "ceph-radosgw@rgw.{{ ansible_hostname }}.{{ item.instance_name }}"
        state: stopped
        enabled: no
      failed_when: false
      with_items: "{{ rgw_instances }}"
```
And I tried to figure out what the item is, then I added a debug task upon it,  rerun the playbook, and the output was below:
```
TASK [debug] ***********************************************************************************************************************
Thursday 27 August 2020  17:39:31 +0800 (0:00:00.226)       0:00:26.002 ******* 
ok: [ceph01] => (item=[{'instance_name': 'rgw0', 'radosgw_address': AnsibleUndefined, 'radosgw_frontend_port': 8080}]) => 
  msg: '[{''instance_name'': ''rgw0'', ''radosgw_address'': AnsibleUndefined, ''radosgw_frontend_port'': 8080}]'
ok: [ceph02] => (item=[{'instance_name': 'rgw0', 'radosgw_address': AnsibleUndefined, 'radosgw_frontend_port': 8080}]) => 
  msg: '[{''instance_name'': ''rgw0'', ''radosgw_address'': AnsibleUndefined, ''radosgw_frontend_port'': 8080}]'
ok: [ceph03] => (item=[{'instance_name': 'rgw0', 'radosgw_address': AnsibleUndefined, 'radosgw_frontend_port': 8080}]) => 
  msg: '[{''instance_name'': ''rgw0'', ''radosgw_address'': AnsibleUndefined, ''radosgw_frontend_port'': 8080}]'
```
The resulst show that the item was still a list,  not a dict, althought it was an item of the loop. So it can not return the value according 'instance_name'. 

What you expected to happen:
I thought this item's value must be a dict, so it could return the instance_name, and the playbook could play normaly.
How to reproduce it (minimal and precise):

Share your group_vars files, inventory and **full** ceph-ansibe log

**Environment**:
* OS (e.g. from /etc/os-release): CentOS Linux release 8.2.2004 (Core)
* Kernel (e.g. `uname -a`): 4.18.0-193.el8.x86_64
* Docker version if applicable (e.g. `docker version`): podman version 1.6.4
* Ansible version (e.g. `ansible-playbook --version`): ansible 2.8.14
* ceph-ansible version (e.g. `git head or tag or stable branch`): stable-4.0.25
* Ceph version (e.g. `ceph -v`): 14.2.11 Nautilus

**Comment by dsavineau** [comment]
```console
{''instance_name'': ''rgw0'', ''radosgw_address'': AnsibleUndefined, ''radosgw_frontend_port'': 8080}
```
Do you see the **AnsibleUndefined** on `radosgw_address` ?
That usually means your ansible group_vars/host_vars aren't loaded correctly. That's why the service task fails with the loop.

> The resulst show that the item was still a list, not a dict, althought it was an item of the loop. So it can not return the value according 'instance_name'.

The `instance_name` can't be return due the AnsibleUndefined in the dict.

Also the `rgw_instances` variable is a list of dict.

Can you share the full ansible-playbook command used ?
Where is located the ansible inventory ?
Where is localted the ansible group_vars/host_vars directories ?

**Comment by xiaohh2016** [comment]
The full ansible-playbook command was:
```
ansible-playbook infrastructure-playbooks/purge-container-cluster.yml
```
and I run this comand in path /root/ceph-ansible/, default inventory was `/etc/ansible/hosts`, the `group_vars` directory was 
inside `/root/ceph-ansible/`。 

In group_vars/all.yml file, I set the radowsgw_address_block to 10.251.48.0/23, but in the previous task set_radosgw_address I debug it with this task:
```
---
- debug:
    msg: "radosgw_address_block is {{ radosgw_address_block }}, ip_version is {{ ip_version }}, radosgw_address is {{ radosgw_address }}, radosgw_interface is {{ radosgw_interface }}"
- name: set_fact _radosgw_address to radosgw_address_block ipv4
.....
```
the output was below:
```
ok: [ceph01] => 
  msg: radosgw_address_block is subnet, ip_version is ipv4, radosgw_address is x.x.x.x, radosgw_interface is interface
ok: [ceph02] => 
  msg: radosgw_address_block is subnet, ip_version is ipv4, radosgw_address is x.x.x.x, radosgw_interface is interface
ok: [ceph03] => 
  msg: radosgw_address_block is subnet, ip_version is ipv4, radosgw_address is x.x.x.x, radosgw_interface is interface
```
which means this task run with default settings in `/root/ceph-ansible/roles/ceph-defaults/defaults/main.yml`, the settings in `all.yml` didn't work at all. Was that means the `pure-container-cluster.yml` didn't use the group vas inside `group_vars/all.yml`？

**Comment by xiaohh2016** [comment]
I found the reason. 
It was because I use the `/etc/ansible/hosts` as the default inventory file to run the `pure-container-cluster.yml` playbook, so the playbook couldn't find `group_vars` under /root/ceph-ansible, so it choosed the `roles/ceph-defaults/defaults/main.yml` as the default vars. So the playbook run with so many errors. 
I solved this problem by two methods:
- copy the full `group_vars` directory to `/etc/ansible/`， so it can find the correct vars in it.
- create a inventory file like `ceph-install` inside `/root/ceph-ansible`, and run the playbook with command
`ansible-playbook infrastructure-playbooks/purge-container-cluster.yml -i ceph-install`
then the playbook will find vars in `/root/ceph-ansible/group_vars`.
